### PR TITLE
Basic Infura Provider Support

### PR DESCRIPTION
**Pull request by KPrasch**
Based on #948

### CODE REVIEW

File: nucypher/blockchain/eth/clients.py [hunk 1]
```diff
@@ -224,6 +228,12 @@ def sign_message(self, account: str, message: bytes) -> str:
 
 class GethClient(Web3Client):
 
+    @classmethod
+    def _get_variant(cls, w3):
+        if 'infura' in w3.provider.endpoint_uri:
```
Comment (KPrasch) [inline nucypher/blockchain/eth/clients.py]: @vepkenez - Some web3 provider instances do not have the `endpoint_uri` attribute

Comment (vepkenez) [reply nucypher/blockchain/eth/clients.py]: you could also do 'if `infura' in getattr(w3.provider, 'endpoint_uri', ''):`
either

**Comment by vepkenez** [comment]
oops I didn't mean to take this out of "draft"...  no way to put it back!

File: nucypher/blockchain/eth/clients.py (hunk 1 above)
Comment (vepkenez) [inline nucypher/blockchain/eth/clients.py]:
```suggestion
        if 'infura' in getattr(w3.provider, 'endpoint_uri', ''):
```

Comment (KPrasch) [reply nucypher/blockchain/eth/clients.py]: :heavy_check_mark:

**Comment by KPrasch** [comment]
Combined with #1071

**Pull request closed without merging**